src/build.c

```diff
@@ -977,26 +964,19 @@ static GPid build_run_cmd(GeanyDocument *doc, guint cmdindex)
 
 		if (vc->skip_run_script)
 		{
-			SETPTR(vte_cmd_nonscript, utils_get_utf8_from_locale(vte_cmd_nonscript));
-			vte_cmd = g_strconcat(vte_cmd_nonscript, "\n", NULL);
-			g_free(vte_cmd_nonscript);
+			SETPTR(run_cmd, utils_get_utf8_from_locale(run_cmd));
+			vte_cmd = g_strconcat(run_cmd, "\n", NULL);
 		}
 		else
-			vte_cmd = g_strconcat("\n/bin/sh ", RUN_SCRIPT_CMD, "\n", NULL);
+			vte_cmd = g_strconcat("\n/bin/sh ", run_cmd, "\n", NULL);
```

Comment:
I think it would be best to

1. Unify the encodings in prepare_run_script() as you mentioned in the other comment
2. Rename prepare_run_script() to prepare_run_cmd()
3. Add "\n/bin/sh " already in prepare_run_cmd() based on vc->skip_run_script so the if test will disappear here.

I'll prepare the fix.